### PR TITLE
feat(metrics): add chain info metrics

### PR DESCRIPTION
**Pull request by kmd-fl**
## Description
Add a chain info metric with chain configuration

## Proposed Changes
- Refactor a bit to remove `server-config` dependency on `peer-metrics`. It turned out to be useless in this PR, but maybe will be helpful in the next one with more configs
- Add `current_commitment_status`. The status is a number which corresponds to the CCStatus enum in chain_listener.
- Add `nox_chain_info` metric. Example:
```
# HELP nox_chain Chain Nox Info.
# TYPE nox_chain info
nox_chain_info{peer_id="0x123",http_endpoint="http://localhost:8545",diamond_contract_address="0x123",network_id="1",default_base_fee="1",default_priority_fee="1",ws_endpoint="ws://localhost:8546",proof_poll_period="1s",min_batch_count="1",max_batch_count="1",max_proof_batch_size="1",epoch_end_window="1s"} 1
# EOF
```

### CODE REVIEW

File: crates/peer-metrics/src/info.rs [hunk 1]
```diff
@@ -17,20 +17,118 @@
...
 use prometheus_client::registry::Registry;
 
+pub struct NoxInfo  {
+    pub versions: NoxVersions,
+    pub chain_info: ChainInfo,
+}
+
+
+#[derive(Debug, Clone, Hash, Eq, PartialEq, EncodeLabelSet)]
+pub struct NoxVersions {
```
Comment (folex) [inline crates/peer-metrics/src/info.rs:31]: imo singular "version" suits better

File: crates/peer-metrics/src/info.rs [hunk 2]
```diff
@@ -17,20 +17,118 @@
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.
  */
 
+use prometheus_client::encoding::EncodeLabelSet;
 use prometheus_client::metrics::info::Info;
 use prometheus_client::registry::Registry;
 
+pub struct NoxInfo  {
+    pub versions: NoxVersions,
```
Comment (folex) [inline crates/peer-metrics/src/info.rs:25]: was expecting array here

File: crates/peer-metrics/src/info.rs [hunk 3]
```diff
@@ -17,20 +17,118 @@
...
+    pub ws_endpoint: String,
+    pub proof_poll_period_secs: u64,
+    pub min_batch_count: usize,
+    pub max_batch_count: usize,
+    pub max_proof_batch_size: usize,
+    pub epoch_end_window_secs: u64,
+}
+
+impl ChainInfo {
+    pub fn empty(peer_id: String) -> ChainInfo {
```
Comment (folex) [inline crates/peer-metrics/src/info.rs:57]: default?

File: crates/peer-metrics/src/info.rs [hunk 4]
```diff
@@ -17,20 +17,118 @@
...
+    let sub_registry = registry.sub_registry_with_prefix("nox");
+
+    let info = Info::new(nox_info.versions);
+    sub_registry.register("build", "Nox Info", info);
+
+    let chain_info = Info::new(nox_info.chain_info);
+    sub_registry.register("chain", "Chain Nox Info", chain_info);
+}
+
+pub fn add_info_metrics2(
```
Comment (folex) [inline crates/peer-metrics/src/info.rs:88]: why 2? maybe remove 1st?

**Review by folex — APPROVED** [review]
cc status in the next pr?

File: crates/peer-metrics/src/info.rs (hunk 2 above)
Comment (kmd-fl) [reply crates/peer-metrics/src/info.rs:25]: Why? `NoxVersions` implement the `EncodeLabelSet` which creates the array automatically from the structure fields and values

File: crates/peer-metrics/src/info.rs (hunk 3 above)
Comment (kmd-fl) [reply crates/peer-metrics/src/info.rs:57]: Well, maybe. I would expect that `default` is what implements `Default`, and I'm not sure if this thing should implement default, but whatever

Comment (folex) [reply crates/peer-metrics/src/info.rs:57]: I think it's a good idea to move `empty` to Default implementation, that's the rust way of things

**Pull request merged**